packages/payer-image/PayerImage.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useEffectAsync } from '@availity/hooks';
+import { avSlotMachineApi } from '@availity/api-axios';
+import get from 'lodash.get';
+
+const spaceIDQuery = `
+query($id: ID!){
+  space(id: $id){
+    images{
+      name
+      value
+    }
+  }
+}
+`;
+
+const payerIDQuery = `
+query($payerIDs: [String!], $types: [String!]){
+  spaces(payerIDs: $payerIDs, types: $types){
+    spaces{
+      images{
+        name
+        value
+      }
+    }
+  }
+}
+`;
+
+const fetchImage = async (query, variables, path, clientId, imageType) => {
+  try {
+    const {
+      data: { data },
+    } = await avSlotMachineApi.create(
+      {
+        query,
+        variables,
+      },
+      { headers: { 'X-Client-ID': clientId } }
+    );
+
+    const images = get(data, path, []).reduce((accum, { name, value }) => {
+      accum[name] = value;
+      return accum;
+    }, {});
+
+    return images[imageType];
+  } catch (error) {
+    return error;
+  }
+};
+
+const getImage = async (spaceId, payerId, clientId, imageType) => {
+  try {
+    if (!clientId) {
+      throw new Error('clientId is required');
+    }
+
+    let url;
+    if (spaceId) {
+      const variables = { id: spaceId };
+      const path = 'space.images';
+      url = await fetchImage(
+        spaceIDQuery,
+        variables,
+        path,
+        clientId,
+        imageType
+      );
+    } else if (payerId) {
+      const variables = { payerIDs: [payerId], types: ['space'] };
+      const path = 'spaces.spaces[0].images';
+      url = await fetchImage(
+        payerIDQuery,
+        variables,
+        path,
+        clientId,
+        imageType
+      );
+
+      // We can probably remove this at some point once our spaces data is complete
```

Comment:
It would be nice for this to be the case now and just name this package `spaces-image` and be slightly more specific to what it's going and more generic for what images it can return (not every space is a payer space 😉)

Reply:
07d0681a52b5d6d8cda9efb584ca37d838f61aca